corehq/apps/reports/util.py

```diff
@@ -442,11 +442,18 @@ def get_installed_custom_modules():
 
 
 def is_mobile_worker_with_report_access(couch_user, domain):
-    return (
-        couch_user.is_commcare_user
-        and domain is not None
-        and Domain.get_by_name(domain).default_mobile_worker_redirect == 'reports'
-    )
+    if not couch_user.is_commcare_user():
+        return False
+
+    if domain is not None:
+        role = couch_user.get_role(domain)
```

Comment:
i looked for that and wondered why that method wasn't a thing. guess i passed over it https://github.com/dimagi/commcare-hq/pull/13268/commits/235aad1506d3d3cd64e2436ccb36a5c1d356c9bf